examples/hotrod/pkg/log/spanlogger.go

```diff
@@ -16,15 +16,244 @@
 package log
 
 import (
+	"fmt"
 	"time"
 
 	"github.com/opentracing/opentracing-go"
 	tag "github.com/opentracing/opentracing-go/ext"
 	"github.com/opentracing/opentracing-go/log"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/trace"
 	"go.uber.org/zap"
 	"go.uber.org/zap/zapcore"
 )
 
+type otelSpanLogger struct {
+	logger     *zap.Logger
+	span       trace.Span
+	spanFields []zapcore.Field
+}
```

Comment:
Another thing that's worth looking into is this: when the Context contains an OT span generated by OTEL Bridge, is there a way to access the real OTEL Span that it wraps? Because if so, rather than maintaining two implementations here we can simply upgrade this spanLogger to only work with OTEL API.

Reply:
Namely, if they don't have it already, I recommend creating a PR for the bridge to implement an Unwrap method:
```go
func Unwrap(opentracing.Span) (trace.Span, error)
```

https://github.com/open-telemetry/opentelemetry-go/blob/dc187e74af528ce081811c4eae17a7930be1232b/bridge/opentracing/bridge.go#L89

Reply:
Doesn't look like they have a public method for that https://pkg.go.dev/go.opentelemetry.io/otel/bridge/opentracing

Reply:
It looks like we can keep a single spanLogger because when writing the span to the context, its tracer is obtained and, because it's a bridge tracer, it's able to inject the span in the "OTEL-way" with the correct context key and OTEL span interface. See: https://github.com/opentracing/opentracing-go/blob/master/gocontext.go#L14.

I've opened another PR that takes advantage of this fact: https://github.com/jaegertracing/jaeger/pull/4556.